### ISSUE TITLE
Duplicated log ingestion

### ISSUE THREAD

**Issue by jack1902** [grafana/k8s-monitoring-helm#104]
TL;DR additional rules can result in duplicate log ingestion or duplication of settings.

If i use the defaults of the chart along with some additional rules, i can double ingest my logs which can result in a log entry being parsed correctly (via my own rules) and incorrectly (via the defaults)[Not the fault of the chart, impossible to know what log format every container in a cluster is pushing out as].

I like the "capture all logs" piece provided by the chart as it ensures i don't overlook a namespace or container in the entire cluster, however if i want to prevent log duplication, i have to add some snippets `logs.pod_logs.extraRelabelingRules` to ensure those pods are dropped, but in doing so i can no longer use `discovery.relabel.pod_logs.output` as an input to another process since pods pods are now dropped entirely meaning there are no valid targets to be selected from in the first place.

The best middle ground i can think of is to have the following flow:
- `discovery.relabel.pod_logs_labelled.output` - this contains all targets in the cluster with additional labels baked in such as `container`, `namespace` etc
- `discovery.relabel.pod_logs.output` - name needs work but basically this contains the `targets` as filtered out by any `extraRelabelingRules` specified by the user

By having this middle ground approach, all logs can be capture by default or particular `targets` can be filtered out of the "catch all" and can be selected by inverse filters by using the `discovery.relabel.pod_logs_labelled.output` which ensures that any "default" labels that are added in future to the chart are not overlooked (i'm duplicating the `discovery.relabel.pod_logs` currently but i'd rather avoid that if possible)

**Comment by petewall** [comment]
This is a big issue and one that I've known about for a while. I will be meeting with some colleagues next Monday to discuss some options around this.

**Comment by petewall** [comment]
So, a few changes have happened with this chart in the past three months!
The default behavior for gathering pod logs is still to gather logs from every pod.
There are a few methods for tuning that:
* You can use `logs.pod_logs.namespaces` to only include pods in the given list of namespaces
* You can use `logs.pod_logs.extraRelabelingRules` to use rules to drop pods
* You can now set this annotation on a pod: `k8s.grafana.com/logs.autogather=skip` to skip gathering logs from that pod.
* Alternatively to the previous, if you set `logs.pod_logs.discovery=annotation`, then the agent will *only* gather logs from pods that set the annotation `k8s.grafana.com/logs.autogather=gather`.

**Comment by petewall** [comment]
We have some options and the new integrations feature in v2 will deal with this.

**Comment by zegomess** [comment]
@petewall hey Pete, how are you doing?

I was wondering if this is also doable in v2, using the annotation `k8s.grafana.com/logs.autogather=skip` to skip specific pods so no duplication happens?

I'm using chart version `2.0.23` with the following configuration for logs:

```
podLogs:
  enabled: true
  gatherMethod: kubernetesApi
```

This is working cluster-wide, but since some of our applications are instrumented with OTel and send their logs, even using the above annotation is causing the logs to be duplicated, where the application sends logs to the receiver component, but it's also gathered by the pod-logs feature... I can't use the `excludeNamespaces` or `namespaces` configurations, because not all pods for a given namespace are instrumented, so it should be a configuration on the pod level

I was using this pod annotation in v1 and it was working as expected... Not sure if I'm missing something in the documentation, if so, could you point that out for me please?
